src/main/java/org/folio/note/NoteRepositoryImpl.java

```diff
@@ -0,0 +1,205 @@
+import org.z3950.zing.cql.cql2pgjson.FieldException;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.sql.UpdateResult;
+
+@Component
+public class NoteRepositoryImpl implements NoteRepository {
+
+  private static final String NOTE_VIEW = "note_view";
+  private static final String NOTE_TABLE = "note_data";
+
+  private final Logger logger = LoggerFactory.getLogger(NoteServiceImpl.class);
+
+  @Autowired
+  private Vertx vertx;
+
+  @Override
+  public Future<NoteCollection> getNotes(String cqlQuery, int offset, int limit, String tenantId) {
+    logger.debug("Getting notes. new query:" + cqlQuery);
+    Future<NoteCollection> future = Future.succeededFuture(null);
+    return future.compose(o -> {
+      CQLWrapper cql;
+      try {
+        cql = getCQLForNoteView(cqlQuery, limit, offset);
+      } catch (CQL2PgJSONException e) {
+        throw new IllegalArgumentException("Failed to parse cql query");
+      }
+      return getNotes(tenantId, cql);
+    });
+  }
+
+  /**
+   * Saves a note record to the database
+   *
+   * @param note - current Note  {@link Note} object to save
+   */
+  @Override
+  public Future<Note> saveNote(Note note, String tenantId) {
+    initId(note);
+    Future<String> future = Future.future();
+    PostgresClient.getInstance(vertx, tenantId)
+      .save(NOTE_TABLE, note.getId(), note, future.completer());
+
+    return future.map(noteId -> {
+      note.setId(noteId);
+      return note;
+    });
+  }
+
+  /**
+   * Fetches a note record from the database
+   *
+   * @param id id of note to get
+   */
+  @Override
+  public Future<Note> getOneNote(String id, String tenantId) {
+    Future<Note> future = Future.future();
+    PostgresClient.getInstance(vertx, tenantId)
+      .getById(NOTE_VIEW, id, NoteView.class,
+        reply -> {
+          if (reply.succeeded()) {
+            NoteView noteView = reply.result();
+            if (Objects.isNull(noteView)) {
+              future.fail(new NotFoundException("Note " + id + " not found"));
+            } else {
+              future.complete(mapNoteView(noteView));
+            }
+          } else {
+            String error = PgExceptionUtil.badRequestMessage(reply.cause());
+            logger.error(error, reply.cause());
+            if (error == null) {
+              future.fail(reply.cause());
+            } else {
+              future.fail(new BadRequestException(reply.cause().getMessage()));
+            }
+          }
+        });
+    return future;
+  }
+
+  @Override
+  public Future<Void> deleteNote(String id, String tenantId) {
+    Future<UpdateResult> future = Future.future();
+    PostgresClient.getInstance(vertx, tenantId)
+      .delete(NOTE_TABLE, id, future.completer());
+    return future.map(updateResult -> {
+      if(updateResult.getUpdated() == 0){
+        throw new NotFoundException();
```

Comment:
would be nice to have more details in the exception, like name of the entity and its id